src/downloader.ts

```diff
@@ -31,17 +33,26 @@ const download = async (targetDirectory: string, versionRange: string) => {
   return binaryPath
 }
 
-const downloadVersion = async (versionRange: string) => {
-  const version = await getLatestVersionForRange(versionRange)
-  const url = getReleaseURL(version)
-  const res = await fetch(url)
-
-  if (res.body === null) {
-    throw new Error('Could not download Deno')
-  }
-
-  return res.body
-}
+const downloadVersion = async (versionRange: string, logger: Logger) =>
```

Comment:
Totally not a blocker, but personally I think it'd be nicer to keep the retry logic separate from the download logic. You could keep `downloadVersion` as is, and then have `downloadVersionWithRetry` which just something like `pRetry(downloadVersion, { ... })`.

Up to you though!

Reply:
Changed it, which also reset the review.